www/serviceDiscovery.js

```diff
@@ -20,38 +20,52 @@ module.exports = {
      * caller will receive callbacks!
      *
      * @param {string} serviceType
-     *            A valid SSDP service type.
+     *            A valid SSDP service type. (e.g. "urn:schemas-upnp-org:service:ContentDirectory:1", "ssdp:all",
+     *            "urn:schemas-upnp-org:service:AVTransport:1")
      * @param {listenCallback} successCallback
      *            Callback to receive SSDP server answers.
      * @param {errorCallback} errorCallback
      *            Callback to receive error messages.
+     * @param {boolean=} normalizeHeaders
+     *            Set true, if you want capitalized headers. If false, headers will be passed unmodified (default).
+     * @param {number=} readTimeout
+     *            Read timeout in milliseconds. (DEFAULT: 4000) Will send a new "M-SEARCH" request after this time.
      */
-    listen: function (serviceType, successCallback, errorCallback) {
-        cordova.exec(successCallback, errorCallback, 'ServiceDiscovery', 'listen', [serviceType]);
+    listen: function (serviceType, successCallback, errorCallback, normalizeHeaders, readTimeout) {
+        var args = [serviceType];
+
+        if (typeof normalizeHeaders === 'boolean') {
+            args.push(normalizeHeaders);
+        }
+
+        if (typeof readTimeout === 'number') {
+            args.push(readTimeout);
+        }
+
+        cordova.exec(successCallback, errorCallback, 'ServiceDiscovery', 'listen', args);
     },
 
     /**
      * Stops listening for SSDP server discovery answers.
      *
      * You will immediately stop receiving updates to your listener. The background thread will be
-     * stopped within 4 seconds (the read timeout).
+     * stopped after read timeout. (4 seconds per default).
      *
      * It is safe to call this multiple times and before any call to {@link listen}.
      *
-     * @param {stopCallback} successCallback
+     * @param {stopCallback=} successCallback
```

Comment:
that = makes no sense.

Reply:
http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values
This is JSDoc, not JavaDoc. And, yes, this argument is optional, since the callback to successCallback is inevitable, due to implementation details. If you don't need to wait on this, you don't need to give the argument.

I'm sorry, but I find comments phrased like that quite offensive. Would you mind changing your style a little, in the future? Questions instead of statements would be really helpful here, especially, since we don't have non-verbal channels available, here. Thanks!